Public/Src/Cache/ContentStore/Distributed/NuCache/BinManager.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using BuildXL.Cache.ContentStore.Hashing;
+using BuildXL.Cache.ContentStore.Utils;
+
+namespace BuildXL.Cache.ContentStore.Distributed.NuCache
+{
+    internal class BinManager
+    {
+        private const int MaxBins = 64 * 1024;
+
+        private readonly MachineLocation[] _bins;
+        private readonly int _locationsPerBin;
+        private readonly int _entriesPerMachine;
+
+        private MachineLocation[][] _machinesForBins;
+
+        private readonly IContentHasher _contentHasher = ContentHashers.Get(HashType.MD5);
+
+        /// <nodoc />
+        public BinManager(int machinesPerBin, int entriesPerMachine, int amountOfBins)

```

Comment:
>int entriesPerMachine, int amountOfBins [](start = 46, length = 39)

At least in the scheme, I had in mind, these values would not need to be specified. Rather, they would be computed based on the number of active machines.